### ISSUE TITLE
Send NMEA for OpenCPN instead of Mavlink

### ISSUE THREAD

**Issue by colinsauze** [boatd/boatd#147]
APMPlanner2 is of limited use for boat control, it can't show wind or rudder/sail states graphically and doesn't have nautical charts. Write a plugin for sending NMEA to OpenCPN over a serial/xbee link instead. The existing boatd-opencpn script does all of this apart from serial output, but should it be merged into the main project as a plugin?

**Comment by kragniz** [comment]
yeah, I think this would be a good idea. The main question is whether to send raw NMEA strings over the xbee, or doing some form of compression + error checking

**Comment by naturesyouth** [comment]
NMEA + CRC should be fine, its not particularly high data-rate.
Only things to check are seeing if we can also send waypoints or direct commands over NMEA, i can dig out my copy of the spec and have a look for that sort of message.

**Comment by kragniz** [comment]
There's `WPL`, but opencpn doesn't support that as an input
> Used to send routes and waypoints up to a GPS receiver. OpenCPN does not process these sentences as input. 

`OSD` looks like a good sentence to send
> Own Ship Data 
    Heading, degrees true
    Status, A = Data Valid
    Vessel Course, degrees True
    Course Reference
    Vessel Speed
    Speed Reference
    Vessel Set, degrees True
    Vessel drift (speed)
    Speed Units


https://www.opencpn.org/wiki/dokuwiki/doku.php?id=opencpn:opencpn_user_manual:advanced_features:nmea_sentences

**Comment by naturesyouth** [comment]
can we get OpenCPN to send wpl? and osd looks very good

**Comment by kragniz** [comment]
opencpn has features for sending waypoints to autopilots over nmea, which can probably be used for planning waypoints at least https://opencpn.org/wiki/dokuwiki/doku.php?id=opencpn:opencpn_user_manual:options_setting:connections#sending_an_active_route_to_an_autopilot

**Comment by kragniz** [comment]
related https://github.com/boatd/boatd-opencpn/issues/2

**Comment by colinsauze** [comment]
I've had boatd-opencpn sending WPL strings to opencpn ok, think I still need to pull request that code. 
There is support in opencpn to send a waypoint to a GPS but it looks a bit buggy and doesn't work through the standard opencpn connections. Maybe we could look at trying to fix opencpn a bit too.
